edx_exams/apps/lti/views.py

```diff
@@ -12,19 +15,96 @@
 from rest_framework.permissions import IsAuthenticated
 from rest_framework.response import Response
 
-from edx_exams.apps.core.api import get_attempt_by_id, update_attempt_status
+from edx_exams.apps.core.api import (
+    get_attempt_by_attempt_number_and_resource_id,
+    get_attempt_by_id,
+    update_attempt_status
+)
 from edx_exams.apps.core.exceptions import ExamIllegalStatusTransition
 from edx_exams.apps.core.statuses import ExamAttemptStatus
 from edx_exams.apps.lti.utils import get_lti_root
 
+log = logging.getLogger(__name__)
+
 EDX_OAUTH_BACKEND = 'auth_backends.backends.EdXOAuth2'
 
+LTI_PROCTORING_ASSESSMENT_CONTROL_ACTIONS = [
+    'pauseRequest',
+    'resumeRequest',
+    'terminateRequest',
+    'update',
+    'flagRequest',
+]
+
+
+@api_view(['POST'])
+@require_http_methods(['POST'])
+@authentication_classes((Lti1p3ApiAuthentication,))
+@permission_classes((LtiProctoringAcsPermissions,))
+def acs(request):
+    """
+    Endpoint for ACS actions
+
+    NOTE: for now just have the actions LOG what the actions is doing.
+    We can implement proper functionality and tests later once we
+    hear back from our third party proctoring service vendors
+    (i.e. Verificient and Proctortrack).
+
+    Currently, we only support flagging of exam attempts.
+    Other ACS actions (pause, resume, terminate, update) could be implemented
+    in the future if desired.
+    """
+    data = json.loads(request.body)
+
+    # This identifies the proctoring tool the request is coming from.
+    user = data['user']
+
+    # The link to exam the user is attempting
+    resource_id = data['resource_link']['id']
+
+    # Exam attempt number (Note: ACS does not differentiate between launches)
+    # i.e, if a launch fails and multiple subsequent launches occur for
+    # the same resource_link + attempt_number combo, the ACS only perceives
+    # this as one singular attempt.
+    attempt_number = data['attempt_number']
+
+    # ACS action to be performed
+    action = data['action']
+
+    # Data validation: Make sure that the exam attempt is either ongoing or completed.
+    # But not verified, rejected, or expired (no need to flag in these cases)
+    # Therefore, any attempt with status from 'started' to 'submitted' can be flagged
+    VALID_STATUSES = [
+        ExamAttemptStatus.started,
+        ExamAttemptStatus.ready_to_submit,
+        ExamAttemptStatus.timed_out,
+        ExamAttemptStatus.submitted,
+    ]
+
+    attempt = get_attempt_by_attempt_number_and_resource_id(attempt_number, resource_id)
+    if attempt.status not in VALID_STATUSES:
+        # TODO: improve this msg with fields and stuff
+        log.info(
+            f'Attempt cannot be flagged for user {user} in',
+            f'with resource id {resource_id} and attempt',
+            f'number {attempt_number}.'
+        )
+        # NOTE: Do we want to create a new exception in exceptions.py just for this case?
```

Comment:
Just want to highlight this question about making a new exception for this case.

Reply:
You could, but how would you use it? You're checking the condition here, so this is where you'd raise the exception. But this is also the place you'd handle the exception.

I think this would be more useful if you were, say, performing a status transition in the API or something and needed to percolate up an exception to the calling code. We don't have a need for that yet, in my opinion.